urld1e4af3f63_GMIyer_myastro_tgz-pJ8-loader_MatchMakerMainJ8/src/myastro/service/matchmaker/impl/RashiAthipatiKootaService.java

```diff
@@ -0,0 +1,196 @@
+
+	private RashiAthipatiKootaService() {
+		rashiAthipatiDetailsCache = getRashiAthipatiDetails();
+	}
+
+	@Override
+	public MatchResult checkKoota(Rashi boyRashi, Rashi girlRashi) {
+		MatchResult matchResult = MatchResult.ADHAMA;
+
+		GrahaRelation boyToGirlGrahaRelation = rashiAthipatiDetailsCache.get(boyRashi.getRashiAthipati()).get(girlRashi.getRashiAthipati());
+		GrahaRelation girlToBoyGrahaRelation = rashiAthipatiDetailsCache.get(girlRashi.getRashiAthipati()).get(boyRashi.getRashiAthipati());
+
+		if (boyToGirlGrahaRelation.equals(GrahaRelation.FRIEND) && girlToBoyGrahaRelation.equals(GrahaRelation.FRIEND)) {
+			matchResult = MatchResult.UTTAMA;
+		}
+		if ((boyToGirlGrahaRelation.equals(GrahaRelation.FRIEND) && girlToBoyGrahaRelation.equals(GrahaRelation.EQUAL)) || (boyToGirlGrahaRelation.equals(GrahaRelation.EQUAL) && girlToBoyGrahaRelation.equals(GrahaRelation.FRIEND))) {
+			matchResult = MatchResult.UTTAMA;
+		}
+		if ((boyToGirlGrahaRelation.equals(GrahaRelation.FRIEND) && girlToBoyGrahaRelation.equals(GrahaRelation.ENEMY)) || (boyToGirlGrahaRelation.equals(GrahaRelation.ENEMY) && girlToBoyGrahaRelation.equals(GrahaRelation.FRIEND))) {
+			matchResult = MatchResult.MADHYAMA;
+		}
+
+		return matchResult;
+	}
+
+	@Override
+	public String getImportance() {
+		return null;
+	}
+
+	private enum GrahaRelation {
+		FRIEND, EQUAL, ENEMY;
+	}
+
+	private Map<Graha, HashMap<Graha, GrahaRelation>> getRashiAthipatiDetails() {
+
+		Map<Graha, HashMap<Graha, GrahaRelation>> rashiAthipatiDetails = new LinkedHashMap<Graha, HashMap<Graha, GrahaRelation>>();
+		rashiAthipatiDetails.put(Graha.SURYA, getSuryaGrahaRelations());
+		rashiAthipatiDetails.put(Graha.CHANDRA, getChandraGrahaRelations());
+		rashiAthipatiDetails.put(Graha.MANGAL, getMangalGrahaRelations());
+		rashiAthipatiDetails.put(Graha.BUDHA, getBudhaGrahaRelations());
+		rashiAthipatiDetails.put(Graha.BRIHASPATI, getBrihaspatiGrahaRelations());
+		rashiAthipatiDetails.put(Graha.SHUKRA, getShukraGrahaRelations());
+		rashiAthipatiDetails.put(Graha.SHANI, getShaniGrahaRelations());
+		rashiAthipatiDetails.put(Graha.RAHU, getRahuGrahaRelations());
+		rashiAthipatiDetails.put(Graha.KETU, getKetuGrahaRelations());
+		return rashiAthipatiDetails;
+	}
+
+	private HashMap<Graha, GrahaRelation> getSuryaGrahaRelations() {
+		HashMap<Graha, GrahaRelation> grahaRelation = new HashMap<Graha, GrahaRelation>();
+		grahaRelation.put(Graha.SURYA, GrahaRelation.FRIEND);
+		grahaRelation.put(Graha.CHANDRA, GrahaRelation.FRIEND);
+		grahaRelation.put(Graha.MANGAL, GrahaRelation.FRIEND);
+		grahaRelation.put(Graha.BUDHA, GrahaRelation.EQUAL);
+		grahaRelation.put(Graha.BRIHASPATI, GrahaRelation.FRIEND);
+		grahaRelation.put(Graha.SHUKRA, GrahaRelation.ENEMY);
+		grahaRelation.put(Graha.SHANI, GrahaRelation.ENEMY);
+		grahaRelation.put(Graha.RAHU, GrahaRelation.ENEMY);
+		grahaRelation.put(Graha.KETU, GrahaRelation.ENEMY);
+		return grahaRelation;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 71:81, 85:95, 99:109, 113:123, 127:137, 141:151, 155:165, 169:179, 183:193.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.